os/kernel/group/group_create.c

```diff
@@ -321,7 +324,13 @@ int group_initialize(FAR struct task_tcb_s *tcb)
 	irqrestore(flags);
 
 #endif
+#if defined(CONFIG_PREFERENCE) && CONFIG_TASK_NAME_SIZE > 0
+	/* The values of private preference are managed by task group.
+	 * Set a name of task group, tg_name for private preference. */
+
+	strncpy(group->tg_name, tcb->cmn.name, CONFIG_TASK_NAME_SIZE + 1);
```

Comment:
I didn't mean the `CONFIG_TASK_NAME_SIZE + 1`.
I think `strlen(tcb->cmn.name) + 1` is correct, because of security attack.